server/leafnode.go

```diff
@@ -628,7 +628,12 @@ func (s *Server) createLeafNode(conn net.Conn, remote *leafNodeCfg) *client {
 
 		c.mu.Unlock()
 		// Error will be handled below, so ignore here.
-		c.parse([]byte(info))
+		err = c.parse([]byte(info))
```

Comment:
In case of a malformed JSON error from a server we also now report the error with debug mode here and close the connection earlier.